rest-service/manager_rest/storage/resource_models.py

```diff
@@ -1930,6 +1937,12 @@ def set_deployment(self, deployment):
     def set_actual_planned_node_instances(self, num):
         self.actual_planned_number_of_instances = num
 
+    def check_unique_query(self):
+        return self.__class__.query.filter(sql_and(
+            self.__class__.id == self.id,
+            self.__class__._deployment_fk == self._deployment_fk
+        ))
```

Comment:
```suggestion
        return super().check_unique_query().filter(
            self.__class__._deployment_fk == self._deployment_fk
        )
```

then, no need for `sql_and`, and we can take advantage of the superclass' possibly setting other filters (tenant) too